src/page/slot.rs

```diff
@@ -382,7 +385,9 @@ where
         offset: usize,
         free: &F,
     ) -> Option<T> {
-        self.release_with(gen, offset, free, |item| item.and_then(|inner| inner.take()))
+        self.release_with(gen, offset, free, |item| {
+            item.and_then(|inner| inner.take())
```

Comment:
style, take it or leave it: we could use point-free style here:
```suggestion
            item.and_then(Option::take)
```